src/Functions/if.cpp

```diff
@@ -55,24 +56,48 @@ inline void fillVectorVector(const ArrayCond & cond, const ArrayA & a, const Arr
     {
         size_t a_index = 0, b_index = 0;
         for (size_t i = 0; i < size; ++i)
-            res[i] = cond[i] ? static_cast<ResultType>(a[a_index++]) : static_cast<ResultType>(b[b_index++]);
+        {
+            if constexpr (std::is_integral_v<ResultType>)
```

Comment:
The optimization looks generally useful. Two comments:
- it would be interesting to know by how much performance improves. While a new performance test under "tests/performance/*.xml" would be perfect (*), just doing a few measurements locally and reporting the numbers in this PR would also be fine for me. Please also double check degenerate cases like 99% `false` as a condition + 1% `true` which will help the CPU branch predictor and give the `?`-based path at a significant advantage over the new multiplication-based path.
- while the PR message eloquently says the optimization is "trivial", it might not be for every reader. Please add a comment in l. 54 a la "Below code attempts to use a branch-free check (multiply with the condition) for better performance instead of conditional evaluation."

(\*) see e.g. tests/performance/if_*.xml for examples